src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -441,7 +441,15 @@ public function fromQuery($query, $bindings = [])
     public function find($id, $columns = ['*'])
     {
         if (is_array($id) || $id instanceof Arrayable) {
-            return $this->findMany($id, $columns);
+            $results = $this->findMany($id, $columns);
+
+            if ($results instanceof Arrayable) {
+                return (new Collection(
```

Comment:
The `new Collection(...)` here was used to preserve the keys of the array from the result.